Uplift third_party/tt-mlir to  2025-06-13

vmilosevic opened the pull request:
This PR uplifts the third_party/tt-mlir to the

vmilosevic commented:
### List of tt-mlir commits since previous uplift:
79da9666a Make tensor_cache thread safe, update invalid assert (#3740)
71e35d58c Uplift third_party/tt-metal to 02bda8e52e88a96f2d0837f517bc4e600fad4706 2025-06-06 (#3689)
efd5766d4 Removing reshape workaround for dump_device (#3731)
becb0a57e Changing ttrt json handling to improve efficiency (#3535)
a3d3e1fb2 Adding an `ttnn::AllGatherOp` workaround  (#3726)
3246fec6c [CI Hotfix] Downgrade TTNN Wheel for PyKernel Test (#3741)
1369e6bf0 Introduce `Conv2dConfig` in TTNN->EmitC (#3721)
a5f389251 Add convenience getChipDesc method for accessing a specific chip id (#3713)

vmilosevic commented:
<table><tr><th><th>Tests</th><th>Passed ✅</th><th>Skipped ⚠️</th><th>Failed</th></tr><tr><td>TT-Torch Tests</td><td>766 ran</td><td>751 passed</td><td>15 skipped</td><td>0 failed</td></tr></table>

<table><tr><th>Test</th><th>Result</th></tr><tr><td colspan="2">No test annotations available</td></tr></table>

<!-- Summary comment for ["TT-Torch Tests"] by mikepenz/action-junit-report -->

jameszianxuTT commented:
We are still blocked by `a5f389251 Add convenience getChipDesc method for accessing a specific chip id (#3713)` -> I will be looking into the mechanics of the failure to understand it better, I have already narrowed down the changeset that caused it.

jameszianxuTT commented:
Testing a different fix related to chipDescIndices in this [onPR check](https://github.com/tenstorrent/tt-torch/actions/runs/15618825906) and validating uplift up to 
```
commit c9abf401778b3dcdf5e43fed232bb5fd33507b5b (origin/main, origin/HEAD)
Author: Tapasvi Patel <133996364+tapspatel@users.noreply.github.com>
Date:   Thu Jun 12 12:45:40 2025 -0500
```

vmilosevic commented:
### List of tt-mlir commits since previous uplift:
3ddc7163e Fix chipDescIndices to map valid indices into chipDescs (#3758)
e3db34777 #3782: Updated device perf csv file to show which ops are const eval values. (#3783)
b77efdc13 #3776: Fix ttrt perf locations to their correct place in the perf results file. (#3777)
9248850db Fix consteval in canonical emitc path (#3716)
c9abf4017 #3495: Added gather op support for automatic parallelization pass with shardy. (#3677)
fb64911e4 Add missing schema version check for system desc (#3763)
27a4f734f Shard join nodes (#3642)
d50b41b1d Uplift model-explorer to f7550b500b64e0a44f91efda33a52939ec58da3e 2025-06-12 (#3767)
75089f8d1 Build tt-mlir with cmake 3.24 (#3736)
fdde2b4b7 Use github.token instead of GITHUB_TOKEN (#3765)
79da9666a Make tensor_cache thread safe, update invalid assert (#3740)
71e35d58c Uplift third_party/tt-metal to 02bda8e52e88a96f2d0837f517bc4e600fad4706 2025-06-06 (#3689)
efd5766d4 Removing reshape workaround for dump_device (#3731)
becb0a57e Changing ttrt json handling to improve efficiency (#3535)
a3d3e1fb2 Adding an `ttnn::AllGatherOp` workaround  (#3726)
3246fec6c [CI Hotfix] Downgrade TTNN Wheel for PyKernel Test (#3741)
1369e6bf0 Introduce `Conv2dConfig` in TTNN->EmitC (#3721)
a5f389251 Add convenience getChipDesc method for accessing a specific chip id (#3713)
